src/appshell/internal/appshellconfiguration.cpp

```diff
@@ -230,9 +230,14 @@ void AppShellConfiguration::rollbackSettings()
     settings()->rollbackTransaction();
 }
 
-void AppShellConfiguration::revertToFactorySettings(bool keepDefaultSettings, bool notifyAboutChanges) const
+void AppShellConfiguration::revertToFactorySettings(bool keepDefaultSettings, bool notifyAboutChanges)
 {
     settings()->reset(keepDefaultSettings, notifyAboutChanges);
+
+    // Unreset the "First Launch Completed" setting so the first-time launch wizard
+    // does not appear. If more settings need to be unresettable in the future,
+    // we could pass those to settings->reset or add "IsNonResettable" to Settings::Item.
+    setHasCompletedFirstLaunchSetup(true);
```

Comment:
What prompted these changes? After a factory reset I _would_ expect the first launch wizard to appear again.

Reply:
This was a request by @avvvvve in the former PR and there is an open issue about it:
![image](https://github.com/user-attachments/assets/1da6777e-7cbe-4ca0-a4d1-e1148d7aacc7)
So some people think the wizard should not appear.

Reply:
Ah thanks - I missed that message.

My only concern with this implementation is that `revertToFactorySettings` is also called when you do a factory reset from the "Help" menu, in which case @avvvvve I'm reasonably certain we _should_ show the wizard again, right?

In practice your changes don't actually cause a problem here because `keepDefaultSettings = false` means we recursively remove the settings directory, so when you later call `setHasCompletedFirstLaunchSetup(true)`, the setting can't be found and we _do_ show the first launch wizard again.

To me this feels a little bit odd and potentially not very future-proof. Is there maybe somewhere else we can put this  logic so that we only set the value to true when it's absolutely necessary?

Reply:
That's why I put it there - so it be executed both when resetting the preferences from the Preferences dialog and when doing a factory reset via the Help menu. In fact, #13299 is specifically about the factory reset.

Deleting the settings directory is not related here since the settings are kept in an .INI file in another location (at least on Windows) and although the settings have been cleared, as soon as `setHasCompletedFirstLaunchSetup(true)` is called, it will create the setting and the wizard _won't_ appear even when doing a factory reset. Just double-checked it on my machine (again, Windows here).

Since this obviously needs a discussion between you, guys, how about we completely remove it from this PR and once you have a decision, implement it in #13299?

Reply:
#13229 is about `Preferences -> Reset preferences`.

> Just resetting preferences shouldn't cause the app to forget that you're an existing user who's already dismissed the start-up wizard, or what your recent files are. I'd expect that for "Revert to factory settings" in the Help menu.  In fact both commands appear to do the same thing, except that the latter at least has a warning.

I think the way this part is worded might be causing some confusion. What they're saying here is that they **_would_** expect "revert to factory settings" to "cause the app to forget that you're an existing user..." (and thus show the wizard).

> Deleting the settings directory is not related

This is interesting - there may indeed be a difference on Mac in this area. On my end I'm still seeing the first time launch wizard after a factory reset from the Help menu. Regardless, I would recommend dropping this commit from the PR until we know for certain what the desired behaviour is.

Reply:
Just to be extra clear:
After 'Revert to factory settings', show the startup wizard on next launch.
After 'Reset preferences', _**do not**_ show the startup wizard on next launch.

Reply:
I don't know why I thought #13299 was about factory reset. Now that I read that story again, I am wondering where my head was previously. So sorry about that! This actually makes our lives easier. Thanks guys!